### PR TITLE
Allow re-using PostgreSQL container in tests

### PR DESCRIPTION
**Pull request by dominikzalewski**
## Description

Ability to iterate faster on integration test development for PostgreSQL by adding the support for TESTCONTAINERS_REUSE_ENABLE environment variable.

## Additional context and related issues

This is based on previous commit 
https://github.com/trinodb/trino/commit/fbd1de01bba0c896e6e584f26d128452fdf35bc1
by @kokosing, which adds the feature and enables it for SQL Server and MySQL. Based on further comments from @kokosing, enablement of this feature for specific database tests needs to be tested one-by-one as not all test containers seem to digest that change without problems. This justifies adding it iteratively for selected databases based on a need-to-use basis.

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java [hunk 1]
```diff
@@ -162,7 +167,12 @@ public String getJdbcUrl()
     @Override
     public void close()
     {
-        dockerContainer.close();
+        try {
+            cleanup.close();
+        }
+        catch (IOException ioe) {
```
Comment (ssheikin) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java:173]: what kind of exception do you expect?

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java [hunk 2]
```diff
@@ -57,6 +60,8 @@ public class TestingPostgreSqlServer
 
     private final PostgreSQLContainer<?> dockerContainer;
 
+    private Closeable cleanup = () -> {};
```
Comment (hashhar) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java:63]: make final, remove assignment

File: testing/trino-testing-services/src/main/java/io/trino/testng/services/ManageTestResources.java [hunk 3]
```diff
@@ -88,7 +89,10 @@ private static boolean isEnabled()
         if (System.getProperty("ManageTestResources.enabled") != null) {
             return Boolean.getBoolean("ManageTestResources.enabled");
         }
-        if (System.getenv("DISABLE_REPORT_RESOURCE_HUNGRY_TESTS_CHECK") != null) {
+        if (getenv("DISABLE_REPORT_RESOURCE_HUNGRY_TESTS_CHECK") != null) {
+            return false;
+        }
+        if (getenv("TESTCONTAINERS_REUSE_ENABLE") != null) {
             return false;
         }
```
Comment (hashhar) [inline testing/trino-testing-services/src/main/java/io/trino/testng/services/ManageTestResources.java:97]: The listener reports a failure when the tests end and doesn't actually prevent a test from running or being green.

Anyway this is a preexisting thing so it needs to be a separate commit with some explanation/justification of the change in the commit message.

Comment (hashhar) [reply testing/trino-testing-services/src/main/java/io/trino/testng/services/ManageTestResources.java:97]: Also FYI @findepi as he's the original author of the check so might have opinions.

**Comment by dominikzalewski** [comment]
github issue for the possibly flaky product test created: https://github.com/trinodb/trino/issues/19189

File: testing/trino-testing-services/src/main/java/io/trino/testng/services/ManageTestResources.java (hunk 3 above)
Comment (hashhar) [inline testing/trino-testing-services/src/main/java/io/trino/testng/services/ManageTestResources.java:97]: nit: if branches can be collapsed to a single one in this case (same return value and same logical operation)

**Review by hashhar — APPROVED** [review]
LGTM

**Pull request merged**